File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Unit/Monster/Model/TheBottom/Looter_Model.ts
import { Mathf } from "../../../../../../../Core/Module/Math/Mathf";
import DungeonManager from "../../../../DungeonManager";
import FileHelper from "../../../../FileHelper";
import { ILoader } from "../../../../Loader/ILoader";
import { SpineLoader } from "../../../../Loader/SpineLoader";
import { MonsterInfo } from "../../MonsterInfo";

export class Looter_Model extends MonsterInfo{
    public static Id = "Looter"

    get MaxHealth(): number {
        if(DungeonManager.Inst.AdvanceLevel >= 7)
            return Mathf.RandomRange(46,50);
        else
            return Mathf.RandomRange(44,48);
    }

    get Loader(): ILoader {
        return new SpineLoader(FileHelper.FormatPath("Monsters/theBottom/looter/skeleton.atlas"),FileHelper.FormatPath("Monsters/theBottom/looter/skeleton.json"),FileHelper.FormatPath("Monsters/theBottom/looter/skeleton.png"));
    }

    get Name(): string {
        return "抢劫的";
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Unit/Monster/Model/TheBottom/Looter_Model.cs
using System.Collections.Generic;
using ModTheSpire.Dungeon;
using ModTheSpire.Power;
using ModTheSpire.Unit.Loader;
using UnityEngine;

namespace ModTheSpire.Unit.Monster.Model.TheBottom
{
    public class Looter_Model : MonsterInfo
    {
        public static string Id = "Looter";
        private ILoader loader;

        public override int MaxHealth => DungeonManager.Inst.AdvanceLevel >= 7 ? Random.Range(46,50) : Random.Range(44,48);

        public override ILoader Loader
        {
            get
            {
                return this.loader ??= new SpineLoader("Monsters/theBottom/looter/skeleton.atlas",
                    "Monsters/theBottom/looter/skeleton.json", "Monsters/theBottom/looter/skeleton.png");
            }
        }

        public override string Name => "抢劫的";

        public Looter_Model(Vector2 pos) : base(pos)
        {
        }
    }
}